### PR TITLE
[fix]: pandas.Index.to_numpy() method's arguments not properly typed

### PR DESCRIPTION
**Pull request by LeMeteore**
After cloning a fresh copy of Pandas.
After generating new stubs using stubtest.
After looking for all the type signature for to_numpy method.

There was nothing to update in the following files:
- ./_libs/tslibs/nattype.pyi
- ./_libs/tslibs/timedeltas.pyi

There was modification to do in the following files:
- ./core/arrays/masked.pyi
- ./core/arrays/numpy_.pyi
- ./core/arrays/base.pyi
- /core/frame.pyi
- ./core/base.pyi

The type signature was removed from the following file:
- ./core/indexes/base.pyi

- [x] Refs #237
- [x] tests added.

### CODE REVIEW

File: pandas-stubs/core/arrays/base.pyi [hunk 1]
```diff
@@ -2,7 +2,11 @@ from typing import Sequence
 
 import numpy as np
 
-from pandas._typing import ArrayLike
+from pandas._typing import (
+    ArrayLike,
+    Scalar,
+    npt as npt,
```
Comment (twoertwein) [inline pandas-stubs/core/arrays/base.pyi:8]: We don't want to re-export `npt`. You can remove `as npt`

**Comment by Dr-Irv** [comment]
@LeMeteore you can see the output from the CI on errors you need to fix.  You also need to add the tests.

File: pandas-stubs/core/arrays/masked.pyi [hunk 2]
```diff
@@ -4,14 +4,22 @@ from pandas.core.arrays import (
     ExtensionOpsMixin,
 )
 
-from pandas._typing import Scalar
+from pandas._typing import (
+    Scalar,
+    npt as npt,
```
Comment (Dr-Irv) [inline pandas-stubs/core/arrays/masked.pyi:9]: remove the `as npt` here and elsewhere

**Review by Dr-Irv — APPROVED** [review]
thanks @LeMeteore

**Pull request merged**